src/Dialogs/Message.cpp

```diff
@@ -153,3 +153,77 @@ ShowMessageBox(const TCHAR *text, const TCHAR *caption, unsigned flags)
 
   return res;
 }
+
+
+
+int    //////// LOCKSCREEN
+ShowLockBox()
+{
+
+  SingleWindow &main_window = UIGlobals::GetMainWindow();
+
+  const unsigned button_width = Layout::Scale(15u);
+  const unsigned button_height = Layout::Scale(15u);
+
+  // Create dialog
+  WindowStyle style;
+  style.Hide();
+  style.ControlParent();
+
+  const DialogLook &dialog_look = UIGlobals::GetDialogLook();
+
+  PixelRect form_rc;
+  form_rc.left = 0;
+  form_rc.top = 0;
+  form_rc.right = button_width;
+  form_rc.bottom = button_height;
+
+  WndForm wf(main_window, dialog_look, form_rc, "", style);
+
+  ContainerWindow &client_area = wf.GetClientAreaWindow();
+  
+  
+//  const int dialog_x = Layout::Scale(44u); 
+//  const int dialog_y = Layout::Scale(44u); 
+  const PixelSize root_size = main_window.GetSize();
+  
+  const int dialog_x = (root_size.cx - button_width) / 4;
+  const int dialog_y = (root_size.cy - button_height) / 4 * 3;
+  wf.Move(dialog_x, dialog_y, button_width, button_height);
+
+  PixelRect button_rc;
+  button_rc.left = 0;
+  button_rc.top = 0; 
+  button_rc.right = button_rc.left + button_width;
+  button_rc.bottom = button_rc.top + button_height;
+
+  // Create buttons
+  WindowStyle button_style;
+  button_style.TabStop();
+
+  StaticArray<Button *, 10> buttons;
```

Comment:
Unsafe C++, prone to memory leaks. Why are there even 10 buttons??
And why are you using `StaticArray` here instead of a plain array?